apis/python/src/tiledbsoma/io/ingest.py

```diff
@@ -1473,14 +1496,144 @@ def update_var(
     )
 
 
+def update_uns(
+    exp: Experiment,
```

Comment:
I know this is a draft PR, but I don't want you to have to do time-consuming rework.

As in our internal design document, this _must_ take a URI not an experiment opened for write. Even if it works for you for step 1 of 5 of planned `update_uns` work. I'm happy to review previous conversations on this topic and/or the design document, offline.

Reply:
The doc says a function that takes a URI would need a different name (than `update_uns`), to avoid confusion vs. existing `update_{obs,var}`.

If simple cases can be supported using this API (which matches existing `update_{obs,var}`), that seems worthwhile, and doesn't preclude supporting a URI-based helper, down the road, for changes that require multiple `open`s.

That was my thinking, anyway. Can discuss more offline.

Reply:
Thanks @ryan-williams . OK to discuss here as well, for transparency with our stakeholders.

* There are things (documented as above) which cannot be accommodated in a function which takes an experiment already opened for write. So we'll need that second function someday anyway.
* If we make an `update_uns` that takes an experiment opened for write, and later a `update_uns_from_uri` (or whatever name) later, then:
  * They will both have the same semantics, which is "make the TileDB-SOMA `uns` look like this new AnnData `uns` as much as you are capable"
  * We can keep around the `update_uns` and it will need to start throwing, for the things it can't do -- this means users will have to modify their notebooks/pipelines/etc to call the new function
  * Or, we go through a deprecation cycle and ask users to plan for the change, letting them know `update_uns` will be going away -- this means users will have to modify their notebooks/pipelines/etc to call the new function
* Or, we make an `update_uns_from_uri` (or whatever name) which takes an experiment URI and we simply do the opens within that function (as sketched in the above-referenced doc)
  * There will be one function with the semantics "make the TileDB-SOMA `uns` look like this new AnnData `uns` as much as you are capable"
  * The list of things of which it is capable will grow over time
  * No need for two not-quite the same functions, and/or deprecation cycle

@aaronwolen -- thoughts from the product POV?

Reply:
(Had a rewrite of the PR description saved as a draft; just submitted it, for reference)

One note on the above:

> We can keep around the `update_uns` and it will need to start throwing, for the things it can't do, that the other one can do

`update_uns` in this PR already throws for the things it can't do; it won't have to "start throwing." If anything, it can be made to support some cases that it doesn't currently (which don't require a URI / multiple `open`s). Cases that require a URI will necessarily be implemented only in the URI-based version, and would continue to throw in the `update_uns` introduced here.

Agreed that having two similar functions (one taking a URI, and handling a superset of cases) could be confusing. OTOH, it could be useful to refrain from committing to an API for the URI-based version now, in case we end up wanting a different API, when we get around to supporting those cases. This `update_uns` matches analogous `update_{obs,var}`, and handles some proportion of the cases we've heard requests for.

Reply:
The two functions are now `update_uns_by_uri` and `_update_uns` ("private"). I've reworked the PR description as well to reflect that.

Reply:
Thanks!

Reply:
We might have landed in an OK place, that reflects the doc… @johnkerl wdyt? I will rename the functions or fold `_update_uns` into `update_uns_by_uri`, if that is still what you think is best.